tictactoe-game/script.js

```diff
@@ -1,50 +1,192 @@
+		if(!flag){
+			elementArray[i].style.color = "red";
+		}else {
+
+			/** get element by class name from the array passed by reference*/
+			diagonalClass = document.getElementsByClassName(elementArray[i]);
+			
+			/** Style each element identified by class name  row and column*/
+			diagonalClass[0].style.color='red';
+		}
+
 	}
-	
-	// 3rd: check if board is full
-	if(moves>8){
-		return "It's a tie";
+}
+
+function winner(){
+
+/**
+	* Players move untill Five turn 
+	* @return {<boolean>}
+*/	
+	if( moves<5 ) {
+		return false;
 	}
-	// continue game
+	
+/**
+	*	Clear the diagonals and arrays containing class names, for fresh value 
+*/
+	clearArray();
+
+/**
+	*Check each rows , columns and Diagonals are match
+	*Style the matching line to RED
+	*@return {<string>} 
+*/
+	for(var row=0;row<3;row++){
+		/** Get each rows*/
+		rowSquer=document.getElementsByClassName('r'+row);
+		/** Check whather each rows not null and they are match*/
+		if(!isEmpty(rowSquer)){
+			if( rowSquer[0].innerHTML===rowSquer[1].innerHTML &&
+				rowSquer[0].innerHTML===rowSquer[2].innerHTML ) {
+				/**Style the values (X/O) to RED iff they are match*/
+				markWinner(rowSquer,flag);
+				return rowSquer[0].innerHTML + ' is the winner.';
+			}
+		}
+
+		for(var col=0;col<3;col++){
+			/** get each colomns in a given row*/
+			colSquer=document.getElementsByClassName('c'+col);
+			/** Check whather each columns not null and they are match*/
+			if(!isEmpty(colSquer)){
+				if( colSquer[0].innerHTML===colSquer[1].innerHTML &&
+					colSquer[0].innerHTML===colSquer[2].innerHTML ) {
+					/**Style the values (X/O) to RED iff they are match*/
+					markWinner(colSquer,flag);
+					return colSquer[0].innerHTML + ' is the winner.';
+				}
+			}
+
+			/** Get each rows and columns and identify each diagonals values and collect there values */
+			var boardElment=document.getElementsByClassName('r'+row+" c"+col);
+			/** *check r0 c2 ,r1 c1 and r2 c0 diagonal coordinates and collect 
+				* there values together with there class names
+			*/
+			if( ((2-row)===col)){
+				diagonals2.push(boardElment[0].innerHTML);
+				diagonal2ClassName.push('r'+row+" c"+col);
+			}
+			/** *check r0 c0, r1 c1 and  r2 c2 diagonals coordinates and collect there values
+				*together with there class names
+			*/
+			if (row===col){
+				diagonals1.push(boardElment[0].innerHTML);
+				diagonal1ClassName.push('r'+row+" c"+col);
+			}
+		}
+}
+				
+	/**
+		* Check each diagonals contain the same value X or O
+		* Check the board is on Tie State
+		* Style to RED the diagonals, when its match. 
+	*/
+
+	if((diagonals1[0]===diagonals1[1]) && (diagonals1[0]===diagonals1[2])){
+		flag=1;
+		markWinner(diagonal1ClassName,flag);
+		return diagonals1[0] + ' is the winner.';
+	}else if((diagonals2[0]===diagonals2[1]) && (diagonals2[0]===diagonals2[2])){
+		flag=1;
+		markWinner(diagonal2ClassName,flag);
+		return diagonals2[0] + ' is the winner.';
+	}else if (moves>8)
```

Comment:
You should do this check first, along with `moves < 5` so you don't run the logic above unnecessarily.